packages/beta/src/api/dataPoints/dataPointsApi.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  DataPointsAPI as DataPointsAPIStable,
+  ItemsWrapper,
+} from '@cognite/sdk';
+
+import {
+  DatapointsMultiQuery,
+  DatapointAggregates,
+  Datapoints,
+  DatapointsMonthlyGranularityMultiQuery,
+  DatapointAggregate,
+} from '../../types';
+
+export class DataPointsAPI extends DataPointsAPIStable {
+  public retrieve = (
+    query: DatapointsMultiQuery
+  ): Promise<DatapointAggregates[] | Datapoints[]> => {
+    return this.retrieveDatapointsEndpoint(query);
+  };
+
+  protected async retrieveDatapointsEndpoint<
+    T extends DatapointAggregates[] | Datapoints[] =
+      | DatapointAggregates[]
+      | Datapoints[]
+  >(query: DatapointsMultiQuery) {
+    const path = this.listPostUrl;
+    const response = await this.post<ItemsWrapper<T>>(path, {
+      data: query,
+    });
+    return this.addToMapAndReturn(response.data.items, response);
+  }
+
+  /**
+   *
+   * ```js
+   * const monthlyAggregatesData = await client.datapoints.retrieveDatapointMonthlyAggregates({ items: [{ id: 123 }] });
+   * ```
+   */
+  public retrieveDatapointMonthlyAggregates = async (
```

Comment:
instead of copying the implementation here, could you just make a 

```
protected doRetrieveDatapointMonthlyAggregates = async <T extends DatapointAggregates[] = DatapointAggregates[] | Datapoints[]> (
```

then 
```
public retrieveDatapointMonthlyAggregates
```
would just call doRetrieveDatapointMonthlyAggregates(...)
```

same here in beta, you'd just call `doRetrieveDatapointMonthlyAggregates<DatapointAggregates[]>` with your new type